src/Core/src/Platform/Android/BorderShapeDrawable.cs

```diff
@@ -0,0 +1,286 @@
+            {
+                if (_clipPath != null)
+                {
+                    _clipPath.Dispose();
+                    _clipPath = null;
+                }
+            }
+
+            DisposeBorder(disposing);
+
+            base.Dispose(disposing);
+        }
+
+        bool HasBorder()
+        {
+            InitializeBorderIfNeeded();
+
+            return _borderWidth > 0;
+        }
+
+        void InitializeBorderIfNeeded()
+        {
+			if (_borderWidth == 0)
+			{
+				DisposeBorder(true);
+				return;
+			}
+
+            if (_maskPath == null)
+                _maskPath = new Path();
+
+            if (_maskPaint == null)
+            {
+                _maskPaint = new Paint(PaintFlags.AntiAlias);
+                _maskPaint.SetStyle(Paint.Style.FillAndStroke);
+
+                PorterDuffXfermode porterDuffClearMode = new PorterDuffXfermode(PorterDuff.Mode.Clear);
+                _maskPaint.SetXfermode(porterDuffClearMode);
+            }
+
+            if (_borderPaint == null)
+            {
+                _borderPaint = new Paint(PaintFlags.AntiAlias);
+                _borderPaint.SetStyle(Paint.Style.Stroke);
+            }
+        }
+
+        public void UpdateBackgroundColor(Color color)
+        {
+            _backgroundColor = color == Color.Default ? AColor.Transparent : color.ToNative();
+
+            InvalidateSelf();
+        }
+
+        public void UpdateCornerRadius(CornerRadius cornerRadius)
+        {
+            _invalidatePath = true;
+
+            _cornerRadius = cornerRadius;
+
+            InitializeBorderIfNeeded();
+            InvalidateSelf();
+        }
+
+        public void UpdateBorderColor(Color strokeColor)
+        {
+            _invalidatePath = true;
+
+            _borderColor = strokeColor == Color.Default ? AColor.Transparent : strokeColor.ToNative();
+
+            InitializeBorderIfNeeded();
+            InvalidateSelf();
+        }
+
+        public void UpdateBorderWidth(double strokeWidth)
+        {
+            _invalidatePath = true;
+
+            _borderWidth = (float)(strokeWidth * _density);
+
+            InitializeBorderIfNeeded();
+            InvalidateSelf();
+        }
+
+        protected virtual void DisposeBorder(bool disposing)
+        {
+            if (disposing)
+            {
+                if (_maskPath != null)
+                {
+                    _maskPath.Dispose();
+                    _maskPath = null;
+                }
+
+                if (_maskPaint != null)
+                {
+                    _maskPaint.SetXfermode(null);
+                    _maskPaint.Dispose();
+                    _maskPaint = null;
+                }
+
+                if (_borderPaint != null)
+                {
+                    _borderPaint.Dispose();
+                    _borderPaint = null;
+                }
+            }
+        }
+
+        Path GetRoundCornersPath(int width, int height, float borderWidth, CornerRadius cornerRadius)
```

Comment:
This will be removed and will use Microsoft.Maui.Graphics to get the path with corner radius.